### PR TITLE
Create new docker containers with Cuda 11.0.

### PR DESCRIPTION
**Pull request by josephevans**
## Description ##
This PR creates new docker containers for Ubuntu and CentOS with Cuda 11.0. These will be used in CI pipelines in the future.

### CODE REVIEW

File: ci/docker/docker-compose.yml [hunk 1]
```diff
@@ -138,6 +138,8 @@ services:
         BASE_IMAGE: nvidia/cuda:11.0-cudnn8-devel-ubuntu18.04
       cache_from:
         - ${DOCKER_CACHE_REGISTRY}/build.ubuntu_gpu_cu110:latest
+    environment:
+      - CUDA_VERSION=11.0.3
```
Comment (leezu) [inline ci/docker/docker-compose.yml:142]: Please reference the upstream bug https://gitlab.com/nvidia/container-images/cuda/-/issues/92

Comment (josephevans) [reply ci/docker/docker-compose.yml]: It turns out these environment variables are not visible during the build process, from what I can tell. Created a workaround in Dockerfile.build.ubuntu.

Comment (leezu) [reply ci/docker/docker-compose.yml]: Thanks

**Comment by josephevans** [comment]
@mxnet-bot run ci [windows-gpu]

**Comment by josephevans** [comment]
@mxnet-bot run ci [windows-gpu]

**Pull request merged**